### ISSUE TITLE
SelectMany (feature request)

### ISSUE THREAD

**Issue by julian-goldsmith** [tyrotoxin/AsyncEnumerable#23]
I'm currently working on some code that runs batches of requests, and returns an IAsyncEnumerable.  It would be helpful to have a SelectMany extension method, so that I don't have to flatten batches manually.

**Comment by tyrotoxin** [comment]
@julian-goldsmith, that's a good suggestion. Let me know if you need other LINQ extension methods, so i can add them at the same time.

**Comment by tyrotoxin** [comment]
@julian-goldsmith, yeah, NUnit extension still does not work properly for .NET core for some reason. Had to upgrade NuGet packages to the latest version to make .NET Framework tests work at least.
Going to add few more extension methods, and do another release.

**Comment by tyrotoxin** [comment]
@julian-goldsmith, I added the following extension methods with their overloads:
- SelectMany
- Append
- Prepend
- OfType
- Concat
- Distinct
- ToDictionaryAsync
- ToLookupAsync
- AggregateAsync

The new [NuGet package v2.2.0](https://www.nuget.org/packages/AsyncEnumerator/2.2.0) has been published.

**Comment by julian-goldsmith** [comment]
Excellent, thank you tyrotoxin.  I use most of those functions quite frequently, so that should help me quite a lot.
